[Copy] Copy referenced constant strings to clipboard

JXNCTED opened the pull request:
Add the feature of only copying the referenced constant strings from the entry to the clipboard. 

Link to #15

JXNCTED committed:
feat: Add parsed serialized string when cloning

- Add parsed serialized string when cloning. When cloning, the new
onject will have null as parsed serialized string. This commit
adds the parsed serialized string to the cloned object.

Link to DD2480-Group1/jabref#15

JXNCTED committed:
Fix: Setting parsed serialization failed tests

- Setting the parsed serialization will result in hasChanged being
set to true, which will cause the entry to be marked as changed. This
caused differences when the original entry was not serlized.

The fix has added a check of original entry to see if it has
parsed serialization, and if it does, it will be set to the new
entry.

Link to DD2480-Group1/jabref#15

JXNCTED committed:
Feat: Add sanity check for null in ClipBoardManager

To prevent copying null to the clipboard, we add a sanity check for null
before setting the content to the clipboard.

Link to DD2480-Group1/jabref#15

JXNCTED committed:
Merge remote-tracking branch 'origin/fix-for-issue-10872' into issue-15/copy-referenced-str-only

andersblomqvist commented on src/main/java/org/jabref/gui/maintable/MainTable.java:
```diff
@@ -258,8 +258,7 @@ public void copy() {
         if (!selectedEntries.isEmpty()) {
-            List<BibtexString> stringConstants = getStringValues();
-
+            List<BibtexString> stringConstants = database.getDatabase().getUsedStrings(selectedEntries).stream().toList();
```
Use `getStringValues()`, but modify the method to use `getUsedStrings(selectedEntries)`

andersblomqvist reviewed (changes requested):
* Make use of the `getStringValues()` method
* Change the `getStringValues()` method to select the used strings

JXNCTED committed:
Fix: extract method for get used string

Extract the method to obtain the referenced string constants from the
MainTable class.

andersblomqvist reviewed (approved):
LGTM 👍
